File: 2020/P15/problem15a.py
data = [int(x) for x in open('problem15.txt', 'r').read().splitlines()[0].split(',')]

last_spoken = 0
spoken = {}
spoken_diff = -1
for i in range(2020):
    if i < len(data):
        last_spoken = data[i]
        spoken[last_spoken] = i
    else:
        if spoken_diff == -1:
            last_spoken = 0
        else:
            last_spoken = spoken_diff
        if last_spoken in spoken:
            spoken_diff = i - spoken[last_spoken]
        else:
            spoken_diff = -1
        spoken[last_spoken] = i

print(last_spoken)

File: 2023/P15/problem15a.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("problem15.txt")
        .expect("Failed to read file");
    let data = data
        .trim()
        .split('\n')
        .collect::<Vec<_>>();
    let data = data[0]
        .split(',');
    let mut sum = 0;
    for s in data {
        let mut v = 0;
        for c in s.chars() {
            v = ((v + (c as u32)) * 17) % 256;
        }
        sum += v;
    }
    println!("{sum}");
}
